### PR TITLE
Implement Alt hotkey to unite stacks of same type, Implement CTRL split even with selected troop

### PR DESCRIPTION
**Pull request by eos428**
Solves #2407 and #2408 (includes the middle mouse rendition of the ctrl split too)

### CODE REVIEW

File: src/fheroes2/army/army.cpp [hunk 1]
```diff
@@ -825,6 +825,22 @@ void Troops::AssignToFirstFreeSlot( const Troop & troop, const uint32_t splitCou
     }
 }
 
+void Troops::JoinAllTroopsOfType( const Troop & targetTroop )
+{
+    const int totalMonsterCount = GetCountMonsters( targetTroop.GetID() );
+    for ( iterator it = begin(); it != end(); ++it ) {
+        if ( !( *it )->isValid() || ( *it )->GetID() != targetTroop.GetID() )
```
Comment (ihhub) [inline src/fheroes2/army/army.cpp:832]: To improve code readability could we please create a local variable from **it** like this `Troop * troop = *it;`?

File: src/fheroes2/army/army.cpp [hunk 2]
```diff
@@ -825,6 +825,22 @@ void Troops::AssignToFirstFreeSlot( const Troop & troop, const uint32_t splitCou
     }
 }
 
+void Troops::JoinAllTroopsOfType( const Troop & targetTroop )
+{
+    const int totalMonsterCount = GetCountMonsters( targetTroop.GetID() );
```
Comment (ihhub) [inline src/fheroes2/army/army.cpp:830]: We should cache `targetTroop.GetID()` expression.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @eos428 , everything looks good. Just few more corrections required. Could you please take a look?

**Comment by ihhub** [comment]
Hi @eos428 could you please fix merging conflicts?

**Comment by eos428** [comment]
@ihhub done :)

**Comment by ihhub** [comment]
@eos428 thank you so much for implemting this logic!

**Pull request merged**